python/cudf/cudf/_lib/reduce.pyx

```diff
@@ -69,7 +68,7 @@ def reduce(reduction_op, Column incol, dtype=None, **kwargs):
             c_out_dtype
         ))
 
-    py_result = DeviceScalar.from_unique_ptr(move(c_result))
+    py_result = DeviceScalar.from_unique_ptr(move(c_result), dtype=col_dtype)
```

Comment:
I could be wrong, but shouldn't we retrieve the `scale` information from c_result.type().scale() and compute precision accordingly here?

Reply:
Sorry, could you explain what you mean by compute precision?

Reply:
Currently libcudf does not compute precision, and it has to be handled by cudf-python. For simple binops, their precisions are computed in:
https://github.com/rapidsai/cudf/blob/6cab04a684510f827e32820e9da7faa9741190c6/python/cudf/cudf/core/column/decimal.py#L150-L164

From the formulae given in the MSFT link, it should be possible to derive for n rows, the precision should be:
```
# add, sub
p+n-1
# mul:
n * p + n - 1
```
where `p` is the precision from the column. Note that finally it should be clipped to `(0, cudf.Decimal64Dtype.MAX_PRECISION]`.

Reply:
On a second pass - it seems like the `precision` does not affect the function result, since the function returns a host value, and `precision` does not participate in constructing the host value.

Reply:
Right now we just return the host value but in the future we could return a device side scalar, so I think we should construct it with the right precision anyways.